runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -362,11 +362,12 @@ end
 ---@param bufnr number
 ---@param client_id number|nil If nil, then return all of the diagnostics.
 ---                            Else, return just the diagnostics associated with the client_id.
-function M.get(bufnr, client_id)
+---@param predicate function|nil Optional function for filtering diagnostics
```

Comment:
In most other places in the docs we use `(number, optional)` to indicate an optional parameter (rather than `number|nil`. For the sake of consistency I'd suggest you stick with that.

Reply:
hey thanks for the review. The current state of docs seem a little all over the place now. When i tried to grep the `runtime/lua` directory i'm seeing 2 variants: `(optional, <type>)` and `(<type>, optional)`. This file however has a lot of `<type>|nil`, and I chose to use `<type>|nil` specifically because the line directly above already uses that

I'm thinking that this PR is already "consistent" in the sense that it follows the conventions in its surrounding code. Since there are so many inconsistencies we can instead have a separate PR which consolidates the different type definition formats in one go. What do you think? Just my two cents, we can still fix the above types if you insist

Reply:
Staying consistent with the nearest code has highest priority I think, so this is fine. Thanks for the explanation.